test/test-lex.cpp

```diff
@@ -1876,6 +1876,18 @@ TEST_F(test_lex, lex_not_shebang) {
                               offsets_matcher(&input, 2, 3))));
   }
 
+  // BOM must not appear before '#!'.
```

Comment:
Make this a separate `TEST` and put this documentation in the test's name.

Reply:
Shouldn't this belong to - https://github.com/quick-lint/quick-lint-js/blob/6c54dc042d691bbbe6d0c83115b5cd6e907c92bd/test/test-lex.cpp#L1842 as the name suggest `lex_not_shebang`. Any particular reason to move it as a separate `TEST`?

Reply:
> Any particular reason to move it as a separate `TEST`?

Yes. You thought the test was different enough to write a comment, so I think it's different enough to put into its own `TEST` block.

Also, all of the other tests are expecting an `error_unexpected_hash_character`, implying that the `#!` was not parsed as a comment. Your test is parsing the `#!` as a comment.